packages/loader/container-loader/src/container.ts

```diff
@@ -630,7 +630,16 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                     this.logConnectionStateChangeTelemetry(value, oldState, reason),
                 shouldClientJoinWrite: () => this._deltaManager.shouldJoinWrite(),
                 maxClientLeaveWaitTime: this.loader.services.options.maxClientLeaveWaitTime,
-                triggerReconnect: (reason: string) => this._deltaManager.triggerReconnect(reason),
+                triggerConnectionRecovery: (reason: string, retryCount: number, duration: number) => {
+                    // We get here when socket does not receive any ops on "write" connection, including
+                    // its own join op. Attempt some recovery options. Need to collect more data to see
+                    // If they actually do anything good.
+                    if (retryCount <= 10) {
+                        this._deltaManager.submit(MessageType.NoOp, null);
```

Comment:
Why are we sending Noops if no connection is there?

Reply:
We do have valid connection, we do not have join on, so runtime does not know about it, but from DeltaManager & PUSH perspective its totally valid connection